kube-client/src/lib.rs

```diff
@@ -469,6 +469,104 @@ mod test {
         Ok(())
     }
 
+    #[tokio::test]
+    #[ignore] // requires a cluster
+    async fn can_operate_on_pod_metadata() -> Result<(), Box<dyn std::error::Error>> {
+        use crate::{
+            api::{DeleteParams, EvictParams, ListParams, Patch, PatchParams, WatchEvent},
+            core::subresource::LogParams,
+        };
+        use kube_core::{ObjectList, ObjectMeta};
+
+        let client = Client::try_default().await?;
+        let pods: Api<Pod> = Api::default_namespaced(client);
+
+        // create busybox pod that's alive for at most 30s
+        let p: Pod = serde_json::from_value(json!({
+            "apiVersion": "v1",
+            "kind": "Pod",
+            "metadata": {
+                "name": "busybox-kube-meta",
+                "labels": { "app": "kube-rs-test" },
+            },
+            "spec": {
+                "terminationGracePeriodSeconds": 1,
+                "restartPolicy": "Never",
+                "containers": [{
+                  "name": "busybox",
+                  "image": "busybox:1.34.1",
+                  "command": ["sh", "-c", "sleep 30s"],
+                }],
+            }
+        }))?;
+
+        match pods.create(&Default::default(), &p).await {
+            Ok(o) => assert_eq!(p.name_unchecked(), o.name_unchecked()),
+            Err(crate::Error::Api(ae)) => assert_eq!(ae.code, 409), // if we failed to clean-up
+            Err(e) => return Err(e.into()),                         // any other case if a failure
+        }
+
+        // Test we can get a pod as a PartialObjectMeta and convert to
+        // ObjectMeta
+        let pod_metadata: kube_core::ObjectMeta = pods.get_metadata("busybox-kube-meta").await?.into();
+        assert_eq!(
+            "busybox-kube-meta",
+            pod_metadata
+                .name
+                .expect("Expected metadata to have a 'name' field")
+        );
+        assert_eq!(
+            Some((&"app".to_string(), &"kube-rs-test".to_string())),
+            pod_metadata
+                .labels
+                .expect("Expected metadata to have a 'name' field")
+                .get_key_value("app")
+        );
+
+        // Test we can get a list of PartialObjectMeta for pods
+        let p_list = pods.list_metadata(&ListParams::default()).await?;
+
+        // Find only pod we are concerned with in this test and fail eagerly if
+        // name doesn't exist
+        let pod_metadata = p_list
+            .items
+            .into_iter()
+            .find(|p| p.metadata.name.as_ref().expect("Pod must have a name") == "busybox-kube-meta")
```

Comment:
```suggestion
            .find(|p| p.name_any() == "busybox-kube-meta")
```
ought to work.. maybe we need to `impl Resource` for `PartialObjectMeta`

Reply:
you can probably just copy-paste the `impl Resource for Object` setup in `object.rs`

Reply:
this will also allow you to shorten a bunch of your asserts above using`ResourceExt` - but it's also probably the magic sauce that will make `watcher` work for it :-)